client/components/popover/menu-item.jsx

```diff
@@ -50,6 +60,7 @@ export default class PopoverMenuItem extends Component {
 				tabIndex="-1"
 				{ ...omit( this.props, 'icon', 'focusOnHover', 'isSelected' ) }
 				className={ classes }
+				icon={ isExternalLink }
```

Comment:
Let's avoid passing unrecognized attributes to `a` and `button` components because they show in the DOM and/or generate React warnings ("Unknown prop `isExternalLink` on <a> tag").

The `icon` attribute used here should also be avoided:  `<PopoverMenuItem isExternalLink={ false }>` ends up creating `<a icon="false">` which is invalid HTML ([#](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a), [#](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes)) even though it does not cause a React warning.

This should work instead:

```js
let ItemComponent = href ? 'a' : 'button';
const itemProps = omit( this.props, 'icon', 'focusOnHover', 'isSelected', 'isExternalLink' );
if ( isExternalLink && href ) {
    ItemComponent = ExternalLink;
    itemProps.icon = true;
}
```